rust_src/src/data.rs

```diff
@@ -789,4 +791,39 @@ pub fn logcount(value: EmacsInt) -> i32 {
     unsafe { rust_count_one_bits(value as usize) }
 }
 
+/// Set SYMBOL's function definition to DEFINITION, and return DEFINITION.
+#[lisp_fn]
+pub fn fset(mut symbol: LispSymbolRef, definition: LispObject) -> LispObject {
+    let sym_obj = symbol.as_lisp_obj();
+    if sym_obj.is_nil() {
+        // Perhaps not quite the right error signal, but seems good enough.
+        xsignal!(Qsetting_constant, sym_obj);
+    }
+
+    let function = symbol.get_function();
+
+    unsafe {
+        if Vautoload_queue.is_not_nil() && function.is_not_nil() {
+            Vautoload_queue =
+                LispObject::cons(LispObject::cons(sym_obj, function), Vautoload_queue);
+        }
+    }
+
+    if is_autoload(function) {
+        put(symbol, Qautoload, function.as_cons_or_error().cdr());
+    }
+
+    // Convert to eassert or remove after GC bug is found.  In the
+    // meantime, check unconditionally, at a slight perf hit.
+    unsafe {
+        if valid_lisp_object_p(definition) == 0 {
+            emacs_abort();
```

Comment:
We use `panic!` in rust.

Reply:
We should block `emacs_abort` from being ported by bindgen.